src/pineko/theory.py

```diff
@@ -397,6 +408,9 @@ def fk(self, name, grid_path, tcard, pdf):
                 check_scvar_evolve(grid, max_as, max_al, check.Scale.FACT)
         # loading ekos to produce a tmp copy
         with eko.EKO.read(eko_filename) as operators:
+            # Skip the computation of the fktable if the eko is empty
+            if len(operators.mu2grid) == 0:
```

Comment:
```suggestion
            if len(operators.mu2grid) == 0 and check.is_num_fonll(tcard['FNS']) :
```

Reply:
Sure but if is nFONLL I don't want to crash, just continue. 
In the other cases you should crash

Reply:
Wait, continue with what? If it cannot compute the fktable, that is it, there are no other things to do when you ask for fktables

Reply:
With the other datasets.... I don't want `fks` to crash

Reply:
But the other datasets would not have empty ekos so it would not crash. Am I missing something?

Reply:
the command will stop and crash, no?

Reply:
Ok so you mean for the other fk_tables of the same datasets. Ok but then first for DIS only BCDMS and NMCPD have more than one fktable in the yaml file. Second, the fact that an eko is empty is physics related, for instance because for that observable you do not have massive contributions. But then if this is true for one of the two Fktable of NMCPD, for example, it would be true also for the other

Reply:
> But then if this is true for one of the two Fktable of NMCPD, for example, it would be true also for the other

but this might or might not be true for BCDMS, because there you iterate energies and so kinematics

however, what does pineko do when an FK table is empty by physics? can we deal with this situation or do we need an empty grid?

Reply:
It is basically impossible to create empty fktables because if the grid is empty, then pineappl will not give the mugrid and xgrid, but it will pass empty lists. This means that then the ekos will be empty and the fktables computation would just fail.

Reply:
Ok, but then at this point I'm back with the `try ... except` pattern: @andreab1997 is raising a very specific error, that could be caught where needed and avoid failure. Isn't it?

Reply:
This is really confusing... before we had a command `fks` that looped over all the datasets and create the fk-tables. 
I'd like to have this working again, such that with one single command you loop on all the datasets of a theory, 
without having to call each dataset separated. If this is not possible anymore fine.

Reply:
But this was never the case... The command `fks` iterates over all the fktables defined in a yamldb of a *single* dataset

Reply:
Ok I see your point, I am used to call pineko once for each dataset so I did not think about this. Maybe we can just do as @AleCandido is proposing and this should be fine for everyone

Reply:
[025a578](https://github.com/NNPDF/pineko/pull/99/commits/025a5787a1dff76ccf13e78bd936abc4228bc486) is the easiest solution I can think about and accommodate the what happens for nFONLL

Reply:
For me this is even ok but we are giving up having the same kind of check for not nFONLL fktables

Reply:
Yes, because in that case (not nFONLL) I don't see why you should end up with a nice empty grid, so you have better to crash!